### PR TITLE
Ea 1516 rdf xml serialization

### PR DESCRIPTION
**Pull request by gsergiu**

### CODE REVIEW

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/RdfResource.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	this.value = value;
+    }
+    
+    @JacksonXmlProperty(isAttribute=true, localName=XmlConstants.XML_RDF_RESOURCE)
+    public String toString() {
+	return value;
+    }
+    
+    @JsonIgnore
+    public static List<RdfResource> convertToRdfResource(String[] elements) {
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/RdfResource.java]: This method should be moved to an Utility class

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlAgentImpl.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+@JsonPropertyOrder({XmlConstants.XML_SKOS_PREF_LABEL, XmlConstants.XML_SKOS_ALT_LABEL, XmlConstants.XML_SKOS_NOTE, XmlConstants.XML_DC_DATE,
+    	XmlConstants.XML_DC_IDENTIFIER, XmlConstants.XML_DCTERMS_HAS_PART, XmlConstants.XML_DCTERMS_IS_PART_OF, XmlConstants.XML_EDM_BEGIN,
+    	XmlConstants.XML_EDM_END, XmlConstants.XML_EDM_HASMET, XmlConstants.XML_EDM_IS_RELATED_TO, XmlConstants.XML_FOAF_NAME, 
+    	XmlConstants.XML_RDAGR2_BIOGRAPHICAL_INFORMATION, XmlConstants.XML_RDAGR2_DATE_OF_BIRTH, XmlConstants.XML_RDAGR2_DATE_OF_DEATH,
+    	XmlConstants.XML_RDAGR2_DATE_OF_ESTABLISHMENT, XmlConstants.XML_RDAGR2_DATE_OF_TERMINATION, XmlConstants.XML_RDAGR2_GENDER,
+    	XmlConstants.XML_RDAGR2_PLACE_OF_BIRTH, XmlConstants.XML_RDAGR2_PLACE_OF_DEATH, XmlConstants.XML_RDAGR2_PROFESSION_OR_OCCUPATION,
+    	XmlConstants.XML_OWL_SAME_AS})
+public class XmlAgentImpl implements XmlBase {
+    	@JsonIgnore
+    	private eu.europeana.entity.definitions.model.impl.BaseAgent agent;
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlAgentImpl.java]: package name not needed here

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlAgentImpl.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    
+    	public XmlAgentImpl(eu.europeana.entity.definitions.model.impl.BaseAgent agent) {
+    	    	this.agent = agent;
+    	}
+    
+    	@JsonIgnore
+	public String addingAdditionalXmlString(String xml) {
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlAgentImpl.java]: this method should go to serializer class

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlAgentImpl.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+	@JacksonXmlElementWrapper(useWrapping=false)
+	@JacksonXmlProperty(localName = XmlConstants.XML_SKOS_PREF_LABEL)
+	public List<XmlMultilingualString> getPrefLabel() {		
+		return XmlMultilingualString.convertToXmlMultilingualString(agent.getPrefLabel());
+	}
+    	
+	@JacksonXmlElementWrapper(useWrapping=false)
+	@JacksonXmlProperty(localName = XmlConstants.XML_DC_DATE)
+	public List<Object> getDcDate() {
+	    	// TODO: GetDcDate from Agent currently not implemented
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlAgentImpl.java]: we can comment out this method for the time being. We will need to discuss to add the xml mapping to specs

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlBase.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package eu.europeana.entity.web.xml.model;
+
+public interface XmlBase {
+
+    public String addingAdditionalXmlString(String xml);
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlBase.java]: this method should go to Serializer/Utility class... is not part of the data model

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlConceptImpl.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+    	@JsonIgnore
+    	private eu.europeana.entity.definitions.model.impl.BaseConcept concept;
+    
+    	public XmlConceptImpl(eu.europeana.entity.definitions.model.impl.BaseConcept concept) {
+    	    this.concept = concept;
+    	}
+    
+    	@JsonIgnore
+    	public String addingAdditionalXmlString(String xml) {
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlConceptImpl.java]: move to serializer/utility class

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlMultilingualString.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+	return language;
+    }
+    
+    @JacksonXmlText
+    public String getValue() {
+	return value;
+    }
+    
+    @JsonIgnore
+    public static List<XmlMultilingualString> convertToXmlMultilingualString(Map<String, List<String>> values) {
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlMultilingualString.java]: move to utility class

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlMultilingualString.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+	    List<String> entryValues = values.get(language);
+	    for (String entryValue : entryValues) {
+		res.add(new XmlMultilingualString(entryValue, language));
+	    }
+	}
+	return res;
+    }
+    
+    @JsonIgnore
+    public static List<Object> convertToXmlMultilingualStringOrRdfResource(Map<String, List<String>> values) {
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlMultilingualString.java]: move to utility class

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlMultilingualString.java [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+		    res.add(new RdfResource(entryValue));
+		else
+		    res.add(new XmlMultilingualString(entryValue, language));
+	    }
+	}
+	return res;
+    }
+    
+    @JsonIgnore
+    public static List<XmlMultilingualString> convertMapToXmlMultilingualString(Map<String, String> values) {
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlMultilingualString.java]: move to utility class

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlPlaceImpl.java [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    
+    	@JsonIgnore
+    	private eu.europeana.entity.definitions.model.impl.BasePlace place;
+    
+    	public XmlPlaceImpl(eu.europeana.entity.definitions.model.impl.BasePlace place) {
+    	    	this.place = place;
+    	}
+    	
+    	@JsonIgnore
+	public String addingAdditionalXmlString(String xml) {
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlPlaceImpl.java]: move to serializer/utility

File: entity-web/src/main/resources/entity-web-mvc.xml [hunk 11]
```diff
@@ -25,7 +25,10 @@
 					<list>
 						<value>application/json;charset=utf-8</value>
 						<value>application/ld+json;charset=utf-8</value>
-						<value>text/plain;charset=utf-8</value>					
+						<value>text/plain;charset=utf-8</value>
+						<value>application/xml</value>
```
Comment (gsergiu) [inline entity-web/src/main/resources/entity-web-mvc.xml]: move this value to the end of the list

File: pom.xml [hunk 12]
```diff
@@ -56,7 +56,7 @@
 		<version.springBoot>1.3.2.RELEASE</version.springBoot>
 		<version.jackson>2.9.4</version.jackson>
 		<version.codehausJackson>1.9.9</version.codehausJackson>
-		<version.commonsApi>0.2.5-RELEASE</version.commonsApi>
+		<version.commonsApi>0.2.6-SNAPSHOT</version.commonsApi>
```
Comment (gsergiu) [inline pom.xml]: are there any changes made to commonsApi?

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlPlaceImpl.java [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+@JacksonXmlRootElement(localName = XmlConstants.XML_EDM_PLACE)
+@JsonInclude(value = JsonInclude.Include.NON_EMPTY)
+@JsonPropertyOrder({XmlConstants.XML_WGS84_POS_LAT, XmlConstants.XML_WGS84_POS_LONG, XmlConstants.XML_WGS84_POS_ALT, XmlConstants.XML_SKOS_PREF_LABEL,
```
[the remark this answers is not in the document]
Comment (gsergiu) [reply entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlPlaceImpl.java]: field order should be the same as in the entity fields document

File: pom.xml (hunk 12 above)
Comment (DenisKatic) [reply pom.xml]: New rdf xml contants added

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlAgentImpl.java [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+@JsonPropertyOrder({XmlConstants.XML_FOAF_DEPICTION, XmlConstants.XML_SKOS_PREF_LABEL, XmlConstants.XML_SKOS_ALT_LABEL, XmlConstants.XML_SKOS_HIDDEN_LABEL,
+    	XmlConstants.XML_FOAF_NAME, XmlConstants.XML_EDM_BEGIN, XmlConstants.XML_RDAGR2_DATE_OF_BIRTH, XmlConstants.XML_RDAGR2_DATE_OF_ESTABLISHMENT,
+    	XmlConstants.XML_EDM_END, XmlConstants.XML_RDAGR2_DATE_OF_DEATH, XmlConstants.XML_RDAGR2_DATE_OF_TERMINATION, XmlConstants.XML_DC_DATE,
+    	XmlConstants.XML_RDAGR2_PLACE_OF_BIRTH, XmlConstants.XML_RDAGR2_PLACE_OF_DEATH, XmlConstants.XML_RDAGR2_GENDER, 
+    	XmlConstants.XML_RDAGR2_PROFESSION_OR_OCCUPATION, XmlConstants.XML_RDAGR2_BIOGRAPHICAL_INFORMATION, XmlConstants.XML_SKOS_NOTE,
+    	XmlConstants.XML_DCTERMS_HAS_PART, XmlConstants.XML_DCTERMS_IS_PART_OF, XmlConstants.XML_EDM_HASMET, XmlConstants.XML_EDM_IS_RELATED_TO,
+    	XmlConstants.XML_DC_IDENTIFIER, XmlConstants.XML_OWL_SAME_AS})
+public class XmlAgentImpl implements XmlBase {
+    	@JsonIgnore
+    	private BaseAgent agent;
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlAgentImpl.java]: Can we switch to the usage of the interface (Agent)?

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlBase.java [hunk 15]
```diff
@@ -0,0 +1,5 @@
+package eu.europeana.entity.web.xml.model;
+
+public interface XmlBase {
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlBase.java]: can we remove the empty interface?

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlConceptImpl.java [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+@JacksonXmlRootElement(localName= XmlConstants.XML_SKOS_CONCEPT)
+@JsonInclude(value = JsonInclude.Include.NON_EMPTY)
+@JsonPropertyOrder({ XmlConstants.XML_FOAF_DEPICTION, XmlConstants.XML_SKOS_PREF_LABEL, XmlConstants.XML_SKOS_ALT_LABEL, XmlConstants.XML_SKOS_HIDDEN_LABEL,
+    	XmlConstants.XML_SKOS_NOTE, XmlConstants.XML_SKOS_NOTATION, XmlConstants.XML_SKOS_BROADER, XmlConstants.XML_SKOS_NARROWER, XmlConstants.XML_SKOS_RELATED,
+    	XmlConstants.XML_SKOS_BROAD_MATCH, XmlConstants.XML_SKOS_NARROW_MATCH, XmlConstants.XML_SKOS_RELATED_MATCH, XmlConstants.XML_SKOS_CLOSE_MATCH,
+    	XmlConstants.XML_SKOS_EXACT_MATCH, XmlConstants.XML_SKOS_IN_SCHEMA})
+public class XmlConceptImpl implements XmlBase {
+
+    	@JsonIgnore
+    	private BaseConcept concept;
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlConceptImpl.java]: can we switch to interface (Concept)?

File: entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlPlaceImpl.java [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+
+@JacksonXmlRootElement(localName = XmlConstants.XML_EDM_PLACE)
+@JsonInclude(value = JsonInclude.Include.NON_EMPTY)
+@JsonPropertyOrder({XmlConstants.XML_FOAF_DEPICTION, XmlConstants.XML_SKOS_PREF_LABEL, XmlConstants.XML_SKOS_ALT_LABEL, XmlConstants.XML_SKOS_HIDDEN_LABEL,
+    	XmlConstants.XML_WGS84_POS_LAT, XmlConstants.XML_WGS84_POS_LONG, XmlConstants.XML_WGS84_POS_ALT, XmlConstants.XML_SKOS_NOTE, 
+    	XmlConstants.XML_DCTERMS_HAS_PART, XmlConstants.XML_DCTERMS_IS_PART_OF, XmlConstants.XML_EDM_IS_NEXT_IN_SEQUENCE, XmlConstants.XML_OWL_SAME_AS})
+public class XmlPlaceImpl implements XmlBase {
+    
+    	@JsonIgnore
+    	private BasePlace place;
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/xml/model/XmlPlaceImpl.java]: Can we switch to the usage of the interface (Place)?

**Pull request merged**